### ISSUE TITLE
Command execution interface support

### ISSUE THREAD

**Issue by ManuelSchneid3r** [wez/wezterm#6008]
Please see [this](https://github.com/albertlauncher/albert/issues/1416). Comments are welcome.

**Comment by wez** [comment]
The linked issue is closed with a number of outstanding items todo.
I've been loosely following the discussions on the mentioned spec for a very long time.

If someone wants to produce a PR that shows the actual implementation impact, I'm open to discussing whether we should proceed with that PR, but I have no personal desire to wade into this or make changes to facilitate this at this time.

**Comment by ManuelSchneid3r** [comment]
The only requirement is [this line](https://github.com/GNOME/gnome-terminal/commit/f1814cb7ecfe6a5cb4f629dcea56db0dec10a242#diff-12f904502f89b1d16bd194c60d6acf7e39701b27b2ea7abc8a2038d9b7f70374). Gnome temrinal does however a lot of other stuff to check if it is set as default temrinal. Probably to let the user choose to make it the default one. However this is not strictly necessary.

**Comment by ManuelSchneid3r** [comment]
@Vladimir-csp `wezterm cli spawn --`does the job. Will `X-TerminalArg=cli spawn --` work?

**Comment by Vladimir-csp** [comment]
Current spec proposal assumes value to be a single argument, so `cli spawn --` would be transmitted as single argument which would not make sense. What is the reason for this complexity? Most terminals use 1 or even 0 args to denote command to execute.

**Comment by ManuelSchneid3r** [comment]
@Vladimir-csp my answer addressing design is out of scope here. I [posted](https://github.com/Vladimir-csp/xdg-terminal-exec/issues/75#issuecomment-5080544200) it in the issue I opened for this particular purpose.

**Comment by bew** [comment]
Hello guys!

I've been exploring the space of terminal startup from desktop file today, and how different people need different conflicting things; aka there is no single solution that's easy to setup to please everyone ><
I ended up looking at other terminals and found that Kitty's use `X-Terminal*` fields, and finally found this issue and the WIP spec.

> [!NOTE]
> ### Few related issues/PRs to ensure I don't loose track of this issue again ^^'
> - #2271
> - #2103, for which the fix added `--cwd .` in the desktop file (breaking workflows that depend on e.g `default_cwd` config)
> - PRs like #5002, #6148 & #7338
> - #1712 & #4275

👉 I'm open to accept a PR to add extra fields to Wezterm's desktop entry, if you're willing to get this moving here 🤔

---

QUESTION: Have you tried already to getting some early support for this in Dolphin/KIO or other DE' filemanagers / terminal-opening features ? (so we avoid breaking users of fix for #2103)

**Comment by ManuelSchneid3r** [comment]
@bew no. I'm almost exclusively interested in the metadata in the desktop entry file.
